src/promql/src/extension_plan/normalize.rs

```diff
@@ -0,0 +1,341 @@
+// Copyright 2022 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::any::Any;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::task::{Context, Poll};
+use std::time::Duration;
+
+use datafusion::arrow::compute;
+use datafusion::common::{DFSchemaRef, Statistics};
+use datafusion::logical_expr::{LogicalPlan, UserDefinedLogicalNode};
+use datafusion::physical_expr::PhysicalSortExpr;
+use datafusion::physical_plan::metrics::{BaselineMetrics, ExecutionPlanMetricsSet, MetricsSet};
+use datafusion::physical_plan::{
+    DisplayFormatType, ExecutionPlan, Partitioning, RecordBatchStream, SendableRecordBatchStream,
+};
+use datatypes::arrow::array::{ArrowPrimitiveType, PrimitiveArray, TimestampMillisecondArray};
+use datatypes::arrow::datatypes::{SchemaRef, TimestampMillisecondType};
+use datatypes::arrow::error::Result as ArrowResult;
+use datatypes::arrow::record_batch::RecordBatch;
+use futures::{Stream, StreamExt};
+
+type Millisecond = <TimestampMillisecondType as ArrowPrimitiveType>::Native;
+
+const TIMESTAMP_COLUMN_NAME: &str = "timestamp";
```

Comment:
We have a special [key](https://github.com/GreptimeTeam/greptimedb/blob/bc9a46dbb70f0b986c156173f9801a1e86d689a9/src/datatypes/src/schema/column_schema.rs#L29) to mark whether a column is a timestamp, could you use it?